ehsm_kms_service/secret_manager_apis.js

```diff
@@ -265,9 +265,104 @@ const createSecret = async (res, appid, payload, DB) => {
     }
 }
 
+//stores the secret value of a new version into a secret object.
+const putSecretValue = async (res, appid, payload, DB) => {
+    try {
+        //get and check Param in payload
+        let secretName = getParam_String(payload, 'secretName')
+        let secretData = getParam_String(payload, 'secretData')
+        const createTime = new Date().getTime()
+        let sm_masterKey
+        let encryptionKeyId
+        let versionId
+        if (!checkStringParam(secretData, true, SECRETDATA_LENGTH_MAX)) {
+            res.send(_result(400, `secretData cannot be empty, must be string and length not more than ${SECRETDATA_LENGTH_MAX}`))
+            return
+        }
+        if (!checkStringParam(secretName, true, SECRETNAME_LENGTH_MAX)) {
+            res.send(_result(400, `secretName cannot be empty, must be string and length not more than ${SECRETNAME_LENGTH_MAX}`))
+            return
+        }
+
+        //query SM_Masterkey and encrypt secretData
+        const query_encryptionKeyId = {
+            selector: {
+                appid,
+                secretName,
+            },
+            fields: ['encryptionKeyId'],
+            limit: 1,
+        }
+        const secret_metadata_res = await DB.partitionedFind('secret_metadata', query_encryptionKeyId)
+        if (secret_metadata_res.docs.length == 0) {
+            encryptionKeyId = ""
+        } else {
+            encryptionKeyId = secret_metadata_res.docs[0].encryptionKeyId
+        }
+        retCMK = await get_SM_Masterkey(DB, appid, encryptionKeyId)
+        if (retCMK.isFind) {
+            sm_masterKey = retCMK.sm_masterKey
+        } else {
+            res.send(_result(400, retCMK.msg))
```

Comment:
If the user uses a new secretName to put, we still return CMK not find? We should confirm the secretName in DB first.

Reply:
no direct update function in CouchDB, can only be replaced by insert

Reply:
problem fixed. now query the secretname first, and then find the encryptionKeyId exists